rig-core/src/providers/aliyun/embedding.rs

```diff
@@ -0,0 +1,239 @@
+            _ => 2048,
+        }
+    }
+
+    /// Validates if the document list meets the model's constraints
+    ///
+    /// # Validation Checks
+    /// 1. Number of documents doesn't exceed model's maximum capacity
+    /// 2. Each document's token count is within the model's token limit
+    ///
+    /// # Returns
+    /// * `Ok(())` if validation passes
+    /// * `Err(EmbeddingError)` with appropriate error message if validation fails
+    fn validate_documents(&self, documents: &[String]) -> Result<(), EmbeddingError> {
+        const AVG_CHARS_PER_TOKEN: usize = 4;
+
+        if documents.len() > self.max_documents() {
+            return Err(EmbeddingError::ProviderError(format!(
+                "Model {} supports maximum {} documents",
+                self.model,
+                self.max_documents()
+            )));
+        }
+
+        for (i, doc) in documents.iter().enumerate() {
+            let estimated_tokens = doc.len() / AVG_CHARS_PER_TOKEN;
+            if estimated_tokens > self.max_tokens() {
+                return Err(EmbeddingError::ProviderError(format!(
+                    "Document #{} exceeds maximum token limit of {}",
+                    i + 1,
+                    self.max_tokens()
+                )));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl embeddings::EmbeddingModel for EmbeddingModel {
+    const MAX_DOCUMENTS: usize = 25;
+
+    /// Returns the default embedding dimension for the current model
+    ///
+    /// # Returns
+    /// * EMBEDDING_V1: 1536 dimensions
+    /// * EMBEDDING_V2: 1536 dimensions
+    /// * EMBEDDING_V3: 1024 dimensions (can be customized)
+    /// * Unknown models: 0 dimensions
+    fn ndims(&self) -> usize {
+        match self.model.as_str() {
+            EMBEDDING_V1 => 1536,
+            EMBEDDING_V2 => 1536,
+
+            // V3 model defaults to 1024 dimensions
+            // Can be customized to [128, 256, 384, 512, 768, 1024]
+            EMBEDDING_V3 => 1024,
+            _ => 0, // Default to 0 for unknown models
+        }
+    }
+
+    /// Generates embeddings for the provided text documents
+    ///
+    /// # Arguments
+    /// * `documents` - Collection of text documents to embed
+    ///
+    /// # Returns
+    /// * `Result<Vec<Embedding>, EmbeddingError>` - Vector of embeddings or error
+    #[cfg_attr(feature = "worker", worker::send)]
+    async fn embed_texts(
+        &self,
+        documents: impl IntoIterator<Item = String> + Send,
+    ) -> Result<Vec<embeddings::Embedding>, EmbeddingError> {
+        let documents: Vec<String> = documents.into_iter().collect();
+
+        self.validate_documents(&documents)?;
+
+        let request = json!({
+            "model": self.model,
+            "input": documents,
+            "dimension": self.ndims.unwrap_or(self.ndims()),
+            "encoding_format": "float",
+        });
+
+        tracing::info!("{}", serde_json::to_string_pretty(&request).unwrap());
```

Comment:
For tracing, please include a string that describes what's being traced. also this should be a debug trace (a future PR will be tackling refactoring all tracing to be consistent among all modules).